pkg/controller/job/job_controller.go

```diff
@@ -661,6 +683,18 @@ func pastActiveDeadline(job *batch.Job) bool {
 	return duration >= allowedDuration
 }
 
+// pastProgressDeadline checks if job has ProgressDeadlineSeconds field set and if it is exceeded.
+func pastProgressDeadline(job *batch.Job) bool {
```

Comment:
Squash this and `pastActiveDeadline` into one and pass `job.Spec.ProgressDeadlineSeconds` or `job.Spec.ActiveDeadlineSeconds` accordingly.